site/tests/openaire/test_components.py

```diff
@@ -38,6 +41,107 @@ def test_on_publish(
     mocked_session.post.assert_called_once_with(
-        openaire_api_endpoint, data=json.dumps(serialized_record), timeout=10
+        post_endpoint,
+        data=json.dumps(serialized_record),
+        timeout=10,
     )
+
+
+def test_on_delete(
+    running_app,
+    monkeypatch,
+    openaire_record_data,
+    community,
+    create_record,
+    openaire_serializer,
+    mocked_session,
+    openaire_api_endpoint,
+    enable_openaire_indexing,
+):
+    """Test on delete component."""
+
+    # Only for this test, modify record service components
+    monkeypatch.setitem(
+        running_app.app.config,
+        "RDM_RECORDS_SERVICE_COMPONENTS",
+        DefaultRecordsComponents + [OpenAIREComponent],
+    )
+
+    record = create_record(openaire_record_data, community)
+    assert record
+
+    recid = record["id"]
+    records_service.delete_record(system_identity, recid, {})
+
+    serialized_record = openaire_serializer.dump_obj(record.data)
+
+    post_endpoint = f"{openaire_api_endpoint}/feedObject"
+    mocked_session.post.assert_called_once_with(
+        post_endpoint, data=json.dumps(serialized_record), timeout=10
+    )
+
+    mocked_session.delete.assert_called_once_with(
+        openaire_api_endpoint,
+        data=json.dumps(
+            {
+                "originalId": f"10.5281/zenodo.{recid}",
+                "collectedFromId": "opendoar____::2659",
+            }
+        ),
+    )
+
+
+def test_on_restore(
+    running_app,
+    monkeypatch,
+    openaire_record_data,
+    community,
+    create_record,
+    openaire_serializer,
+    mocked_session,
+    openaire_api_endpoint,
+    enable_openaire_indexing,
+):
+    """Test on restore component."""
+
+    # Only for this test, modify record service components
+    monkeypatch.setitem(
+        running_app.app.config,
+        "RDM_RECORDS_SERVICE_COMPONENTS",
+        DefaultRecordsComponents + [OpenAIREComponent],
+    )
+
+    record = create_record(openaire_record_data, community)
+    assert record
+
+    recid = record["id"]
+
+    records_service.delete_record(system_identity, recid, {})
+    records_service.restore_record(system_identity, recid)
+
+    serialized_record = openaire_serializer.dump_obj(record.data)
+
+    post_endpoint = f"{openaire_api_endpoint}/feedObject"
+    # HTTP POST will be called twice (one to create the record and one to restore it)
+    post_calls = [
+        call(post_endpoint, data=json.dumps(serialized_record), timeout=10),
+        call(post_endpoint, data=json.dumps(serialized_record), timeout=10),
+    ]
+
+    # HTTP DELETE will be called once, when the record is deleted
+    delete_calls = [
+        call(
+            openaire_api_endpoint,
+            data=json.dumps(
+                {
+                    "originalId": f"10.5281/zenodo.{recid}",
+                    "collectedFromId": "opendoar____::2659",
+                }
+            ),
+        )
+    ]
+
+    mocked_session.post.assert_has_calls(post_calls)
```

Comment:
does it make sense to check the returns and arguments too ?

Reply:
Returns are completely [mocked](https://github.com/alejandromumo/zenodo-rdm/blob/add_openaire_delete_restore/site/tests/openaire/conftest.py#L29) since we're not hitting any real API. 

Arguments are already being checked since we're passing specific `call` objects: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls